### PR TITLE
Adding composition operator (>>>)

### PR DESCRIPTION
**Pull request by dalkire**
The compose operator ">>" provides an infix notation for function composition.
Composing a function that goes from A -> B with a function that goes from B -> C creates a new function that goes from A -> C.

Also added a tests.

### CODE REVIEW

File: Sources/Prelude/Operators.swift [hunk 1]
```diff
@@ -45,6 +45,14 @@ precedencegroup WayfairPipe {
 /// Definition can be found in /Sources/Prelude.swift
 infix operator |>: WayfairPipe
 
+precedencegroup WayfairCompose {
+    associativity: left
+    higherThan: LogicalConjunctionPrecedence
```
Comment (peter-tomaselli) [inline Sources/Prelude/Operators.swift]: Does this make mixtures of `>>>` and `|>` ambiguous (they both have the same precedence)?

EDIT: I also think (could be wrong) that we want this to be `associativity: right`, as well.

File: Sources/Prelude/Prelude.swift [hunk 2]
```diff
@@ -34,3 +34,10 @@ public func flip<A, B, C>(_ f: @escaping (A, B) -> C) -> (B, A) -> C {
 public func |><A, B>(lhs: A, rhs: (A) -> B) -> B {
     return rhs(lhs)
 }
+
+/// The compose operator ">>>" provides an infix notation for function composition.
+/// Composing a function that goes from A -> B with a function that goes from B -> C
+/// creates a new function that goes from A -> C.
+public func >>><A, B, C>(f: @escaping (A) -> B, g: @escaping (B) -> C) -> ((A) -> C) {
```
Comment (peter-tomaselli) [inline Sources/Prelude/Prelude.swift]: Sorry, annoying style comment. I think my preference here would be to spell the `f` argument `lhs`, and the `g` argument `rhs`, thus emphasizing the operatorness of this function. No worries if you disagree, but it’s what I’ve been trying to do so far.

Also, you should be able to spell the end of this signature without the last outer pair of parens!
```
((A) -> B, (B) -> C) -> (A) -> C { …
```

**Review by peter-tomaselli — CHANGES_REQUESTED** [review]
Looks good! I do think that we need to carefully noodle around with the operator precedences and whatnot to make sure that everything plays well together.

File: Sources/Prelude/Operators.swift (hunk 1 above)
Comment (dalkire) [reply Sources/Prelude/Operators.swift]: You're right. Fixed. Added additional tests.

File: Sources/Prelude/Prelude.swift (hunk 2 above)
Comment (dalkire) [reply Sources/Prelude/Prelude.swift]: Fixed

**Review by peter-tomaselli — APPROVED** [review]

**Pull request merged**